### PR TITLE
add support for DO_CHANGE_ALTITUDE

### PR DESCRIPTION
**Pull request by sfuhrer**
This adds support in PX4 for the MAVLink command [DO_CHANGE_ALTITUDE](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_CHANGE_ALTITUDE.). 

As I implemented it now, it does the exact same thing as DO_REPOSITION that only has the altitude field populated. 

This came up as PX4 is supposed to support all available MAVLink commands. We should actually define it more closely what the expected outcome of the command is (on the MAVLink side), e.g. what happens when the vehicle is currently not in Loiter flight mode. My interpretation is to handle it like DO_REPOSITION, thus switching to Loiter mode first if possible, and then climb/descent to the set altitude. 

We can discuss also supporting other MAV_FRAMEs, or reject commands that have them set to non-absolute altitude amsl.

### CODE REVIEW

File: src/modules/navigator/navigator_main.cpp [hunk 1]
```diff
@@ -370,6 +370,83 @@ void Navigator::run()
 
 				// CMD_DO_REPOSITION is acknowledged by commander
 
+			} else if (cmd.command == vehicle_command_s::VEHICLE_CMD_DO_CHANGE_ALTITUDE
+				   && _vstatus.arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
```
Comment (dagar) [inline src/modules/navigator/navigator_main.cpp:374]: Or should commander reject when disarmed?

Comment (sfuhrer) [reply src/modules/navigator/navigator_main.cpp:374]: Sounds cleaner to me too yes. @bkueng agree?

Comment (bkueng) [reply src/modules/navigator/navigator_main.cpp:374]: Yes

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -718,6 +718,30 @@ Commander::handle_command(const vehicle_command_s &cmd)
...
+
+			} else {
+				cmd_result = vehicle_command_ack_s::VEHICLE_CMD_RESULT_TEMPORARILY_REJECTED;
+				mavlink_log_critical(&_mavlink_log_pub, "Change altitude command rejected\t");
+				/* EVENT
+					* @description Check for a valid position estimate
+					*/
+				events::send(events::ID("commander_change_altitude_rejected"),
+				{events::Log::Error, events::LogInternal::Info},
+				"Change altitude command rejected");
```
Comment (bkueng) [inline src/modules/commander/Commander.cpp:739]: You could simplify this to `printRejectMode(vehicle_status_s::NAVIGATION_STATE_AUTO_LOITER);` (the same for reposition)

File: src/modules/navigator/navigator_main.cpp (hunk 1 above)
Comment (sfuhrer) [reply src/modules/navigator/navigator_main.cpp:374]: I actually actively decided against it originally, see https://github.com/PX4/PX4-Autopilot/pull/19776

Comment (sfuhrer) [reply src/modules/navigator/navigator_main.cpp:374]: I'm in favor of not changing it withing the context of this PR

File: src/modules/commander/Commander.cpp (hunk 2 above)
Comment (sfuhrer) [reply src/modules/commander/Commander.cpp]: done

File: src/modules/navigator/navigator_main.cpp (hunk 1 above)
Comment (bkueng) [reply src/modules/navigator/navigator_main.cpp:374]: Unless I'm missing some use-case I would just reject it while disarmed.

**Review by bkueng — APPROVED** [review]
We can do the logic change separately

File: src/modules/navigator/navigator_main.cpp (hunk 1 above)
Comment (sfuhrer) [reply src/modules/navigator/navigator_main.cpp:374]: Yes let's do that separately.

**Pull request merged**